src/GraphQLMiddleware.php

```diff
@@ -67,12 +61,7 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
     }
 
     private function isGraphQLRequest(ServerRequestInterface $request) {
-        return $this->hasUri($request) || $this->hasGraphQLHeader($request);
```

Comment:
The uri path should not be hard coded, i agree with you on this, but the middleware should know  the name of the uri. Anyone should be able to decide wich uri want to use. To do that, your proposed solution didn't fully work. Because you're hard coding the uri in the `ModuleConfig.php`  in the `path` (the ModuleConfig is zend specific use case, but is not the only, mayby someone want to use the middleware with slim framework).

The middleware should provide a generic solution. To do this there are 2 possibilities:
1) let the user customize the uri via configuration variable
2) let the middleware aware of the `application/graphql` header and handle those request that have this specific header.
 
The actual implementation have, in the `GraphQLMiddleware.php` file, the  `$graphql_uri` variable that actually is hard-coded, but i put that there because i wanted to expand in the future.

To let me accept the PR you should:
1) revert the modification about the uri in GraphQLMiddleware.php
2) add a constructor parameter in GraphQLMiddleware.php, like this

```
public function __construct(Processor $processor, $graphql_uri = "/graphql")
    {
        $this->processor = $processor;
        $this->graphql_uri = $graphql_uri;
    }
```

3) Let the `GraphQLMiddlewareFactory.php` handle the initialitazion

```
 public function __invoke(ContainerInterface $container)
    {
        $processor = $container->get("graphql.processor");

        //read graphql settings
        $graphql_settings = $container->get("graphql");
        
        //do some check
        //...

        return new GraphQLMiddleware($processor, $graphql_settings['uri']);
    }
```

This way, if you're using for example `zend-expressive` and want to point to `/api/graphql` other than `/graphql` you can do that by changing, for example, the config/autoload/global.php`

```
"graphql" => [
        "schema" => MyApp\Schema\MySchema::class,
        "graphql_uri" => "/api/graphql"
    ],
```

If not using zend-expressive you can manually instanziate the middleware

```
....
//init processor
$processor = new Processor($container, $schema);
$uri = "/api/graphql";
$graphqlMiddleware = new GraphQLMiddleware($processor, $uri);

$app->pipe($uri, $graphqlMiddleware);
```